### PR TITLE
Adding strawman threads.

### PR DESCRIPTION
**Pull request by flagxor**

### CODE REVIEW

File: BinaryEncoding.md [hunk 1]
```diff
@@ -553,47 +571,193 @@ operand the index into the table. Its `reserved` immediate is for
...
 
-| Name | Type | Description |
-| ---- | ---- | ---- |
-| flags | `varuint32` | a bitfield which currently contains the alignment in the least significant bits, encoded as `log2(alignment)` |
-| offset | `varuint32` | the value of the offset |
+| Name | Opcode | Immediate | Description |
+| ---- | ---- | ---- | ---- |
+| `i32.sc_load` | `0x28` | 0x08 + `memory_flags` `offset` | atomic load from memory |
+| `i64.sc_load` | `0x29` | 0x08 + `memory_flags` `offset` | atomic load from memory |
+| `f32.sc_load` | `0x2a` | 0x08 + `memory_flags` `offset` | atomic load from memory |
```
Comment (binji) [inline BinaryEncoding.md:608]: FYI: SABs don't have f32 or f64 atomics.

File: BinaryEncoding.md [hunk 2]
```diff
@@ -553,47 +571,193 @@ operand the index into the table. Its `reserved` immediate is for
...
-The `memory_immediate` type is encoded as follows:
+## Atomics
 
-| Name | Type | Description |
-| ---- | ---- | ---- |
-| flags | `varuint32` | a bitfield which currently contains the alignment in the least significant bits, encoded as `log2(alignment)` |
-| offset | `varuint32` | the value of the offset |
+| Name | Opcode | Immediate | Description |
+| ---- | ---- | ---- | ---- |
+| `i32.sc_load` | `0x28` | 0x08 + `memory_flags` `offset` | atomic load from memory |
```
Comment (binji) [inline BinaryEncoding.md:606]: I'd guess we wouldn't want to encode seq. cst. into the opcode name, right?

File: BinaryEncoding.md [hunk 3]
```diff
@@ -553,47 +571,193 @@ operand the index into the table. Its `reserved` immediate is for
...
+| `i32.sc_xor8_u` | `0x2d` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i32.sc_xor16_s` | `0x2e` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i32.sc_xor16_u` | `0x2f` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i64.sc_xor8_s` | `0x30` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i64.sc_xor8_u` | `0x31` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i64.sc_xor16_s` | `0x32` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i64.sc_xor16_u` | `0x33` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i64.sc_xor32_s` | `0x34` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i64.sc_xor32_u` | `0x35` | 0x58 + `memory_flags` `offset` | atomic xor to memory |
+| `i32.sc_swap` | `0x28` | 0x68 + `memory_flags` `offset1` `offset2` | atomic swap to memory |
```
Comment (binji) [inline BinaryEncoding.md:699]: bikeshed: swap makes me think byteswap. why not exchange?

Comment (binji) [inline BinaryEncoding.md:699]: exchange has one offset (cmp_exchange has two)

Comment (binji) [reply BinaryEncoding.md:699]: Oops, I mean they both have one offset, but cmp_exchange has two values

**Comment by flagxor** [comment]
Closing, as Ben's proposal supercedes this.

**Pull request closed without merging**